pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -545,11 +553,11 @@ def _update_best_and_save(
         filepath: str,
         current: torch.Tensor,
         epoch: int,
+        step: int,
```

Comment:
we could read these off the `trainer` directly. the meta question is: we'd always save the trainer.current_epoch and trainer.global_step right? no other values for epoch/step in the checkpoint right? if you want, you could add a small util function in this class 
```
def get_epoch_and_step(trainer) -> Tuple[int, int]:
    return trainer.current_epoch, trainer.global_step
```
then we can use the trainer everywhere and not pass epoch or step just for logging

Reply:
let's make it as separate PR